app/src/main/kotlin/de/rochefort/childmonitor/ListenService.kt

```diff
@@ -0,0 +1,223 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Child Monitor. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.rochefort.childmonitor
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Intent
+import android.content.pm.ServiceInfo
+import android.media.AudioManager
+import android.media.AudioTrack
+import android.media.MediaPlayer
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import android.widget.Toast
+import androidx.core.app.NotificationCompat
+import androidx.core.app.ServiceCompat
+import java.io.InputStream
+import java.net.Socket
+
+class ListenService : Service() {
+    private val frequency: Int = AudioCodecDefines.FREQUENCY
+    private val channelConfiguration: Int = AudioCodecDefines.CHANNEL_CONFIGURATION_OUT
+    private val audioEncoding: Int = AudioCodecDefines.ENCODING
+    private val bufferSize = AudioTrack.getMinBufferSize(frequency, channelConfiguration, audioEncoding)
+    private val byteBufferSize = bufferSize * 2
+    private val binder: IBinder = ListenBinder()
+    private lateinit var notificationManager: NotificationManager
+    private var listenThread: Thread? = null
+    val volumeHistory = VolumeHistory(16384)
+    var childDeviceName: String? = null
+        private set
+
+    override fun onCreate() {
+        super.onCreate()
+        this.notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        Log.i(TAG, "Received start id $startId: $intent")
+        // Display a notification about us starting.  We put an icon in the status bar.
+        createNotificationChannel()
+        intent.extras?.let {
+            val name = it.getString("name")
+            childDeviceName = name
+            val n = buildNotification(name)
+            val foregroundServiceType = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK else 0 // Keep the linter happy
+            ServiceCompat.startForeground(this, ID, n, foregroundServiceType)
+            val address = it.getString("address")
+            val port = it.getInt("port")
+            doListen(address, port)
+        }
+        return START_REDELIVER_INTENT
+    }
+
+    override fun onDestroy() {
+        this.listenThread?.let {
+            this.listenThread = null
+            it.interrupt()
+        }
+
+        // Cancel the persistent notification.
+        val NOTIFICATION = R.string.listening
```

Comment:
this should not be all caps

Reply:
Passed simply `R.string.listening` directly.